### PR TITLE
chore(upgrade): upgrade multiple packages

### PR DESCRIPTION
**Pull request by ThakurKarthik**
Updated multiple packages to their latest versions.
Pushed the code to check the output of the github actions

### CODE REVIEW

**Comment by Shane32** [comment]
Probably need to bump the GitHub Action node version here:

https://github.com/graphql-dotnet/graphql-dotnet/blob/master/.github/workflows/test-documentation.yml#L15

File: .github/workflows/test-documentation.yml
Comment (sungam3r) [reply .github/workflows/test-documentation.yml:15]: Also the same changes here

**Comment by sungam3r** [comment]
I briefly looked through all files except two `.lock` files and left comments. @ThakurKarthik also please pay attention to warnings like these:
```
warning " > graphiql@0.11.11" has incorrect peer dependency "graphql@^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0".
warning " > graphiql@0.11.11" has unmet peer dependency "prop-types@>=15.5.0".
warning "graphiql > codemirror-graphql@0.6.12" has incorrect peer dependency "graphql@^0.10.0 || ^0.11.0".
warning "graphiql > codemirror-graphql > graphql-language-service-parser > graphql-language-service-types@0.1.14" has incorrect 
```

**Comment by sungam3r** [comment]
Side note - labeler did not put `CI` label though file from .github/worflows folder was changed.

**Comment by Shane32** [comment]
> Can we know why the "dependency review" check is failing?

No idea.  I ignore problems by "Dependency Review", "CodeQL analysis", and "Code scanning results".  @sungam3r may know.

**Comment by Shane32** [comment]
@ThakurKarthik Thanks!

> Please pull my changes and give it a test.

@sungam3r Do you want to review this or me?  I don't think we both need to, but one of us should do as suggested and look at the rendered site before merging.

**Comment by Shane32** [comment]
I don't think the root package.json should be needed at all.  It is not used at all for developing, building or deploying the code.  If we look at the documentation scripts we can see that it uses yarn to install the root packages, then uses yarn to install the packages under the docs2 folder.  So as long as the docs2 package json file has everything necessary for building the documentation, I would expect that the root package.json file can be deleted (along with the corresponding actions within the GitHub Action scripts).  Notice also that most of the actions in the package.json file refer to a tools folder which does not exist anymore.

@sungam3r agree?

**Comment by sungam3r** [comment]
> Out of my curiousity how does other devs consume this package?

What package are you talking about? `package.json` file? If so then I'm sure no one uses it. For now repo has two maintainers and we both work with C#. Long time ago a system was created to generate a site with documentation, even before we began to support this project. This generator works on СI. I never tried to launch it and did not look into sources. I'm not a JS developer at all.

**Comment by sungam3r** [comment]
Personally I would move there 5 files (or delete if unused) from the repo root into `docs2` folder
изображение

All JS stuff in this repo is needed ONLY to build documentation site, JS code is not used by GraphQL.NET at all.

**Comment by sungam3r** [comment]
Also I stll see warnings:
```
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning " > graphiql@0.11.11" has incorrect peer dependency "graphql@^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0".
warning " > graphiql@0.11.11" has unmet peer dependency "prop-types@>=15.5.0".
warning "graphiql > codemirror-graphql@0.6.12" has incorrect peer dependency "graphql@^0.10.0 || ^0.11.0".
warning "graphiql > codemirror-graphql > graphql-language-service-parser > graphql-language-service-types@0.1.14" has incorrect peer dependency "graphql@^0.10.0 || ^0.11.0".
[4/4] Building fresh packages...
```

**Comment by sungam3r** [comment]
I don't know the reason why `Dependency Review` failed. You can ignore it since this check is not required one. Test workflow failed on Windows machine at code-first AOT compilation: 
изображение

I rerun the build.

**Comment by sungam3r** [comment]
@Shane32 By the way, new SDK is here https://dotnet.microsoft.com/en-us/download/dotnet/7.0, again.

**Comment by sungam3r** [comment]
@ThakurKarthik It was some transient error.

**Review by Shane32 — APPROVED** [review]
Changes look good, but I didn't actually generate the docs and see what was produced.  @sungam3r to review also.

**Comment by sungam3r** [comment]
> Cool can we merge it?

https://github.com/graphql-dotnet/graphql-dotnet/pull/3585#issuecomment-1505232529

**Comment by Shane32** [comment]
> > Cool can we merge it?
> 
> [#3585 (comment)](https://github.com/graphql-dotnet/graphql-dotnet/pull/3585#issuecomment-1505232529) to review generated site before merge in master.

It is simple to build the docs locally for review, so I'm not sure if this is a comment for yourself, for @ThakurKarthik , for me, or a request that @ThakurKarthik make further changes.  We know they build in the CI environment as the tests pass.  Please clarify.

**Review by sungam3r — APPROVED** [review]

**Comment by Shane32** [comment]
Once merged the updated docs published successfully and should be live now.  Looks good to me! :+1:

**Comment by sungam3r** [comment]
Agree, links seem functional.

**Pull request merged**
